core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -254,8 +254,8 @@ private static Types.MapType mapFromJson(JsonNode json) {
   }
 
   public static Schema fromJson(JsonNode json) {
-    Type type  = typeFromJson(json);
-    Preconditions.checkArgument(type.isNestedType() && type.asNestedType().isStructType(),
+    Type type = typeFromJson(json);
+    Preconditions.checkArgument(type != null && type.isNestedType() && type.asNestedType().isStructType(),
```

Comment:
Can you open a separate PR for Java fixes? This looks good to me, but we try not to mix changes like this into other PRs to avoid commit conflicts when cherry picking to maintain branches.

Reply:
I've actually cherry-picked those changes from https://github.com/apache/iceberg/pull/5276 in this branch. Once that one is in, I'll rebase this branch 👍🏻

Reply:
Sounds good to fix this one, but I think the Java changes should still go in a separate PR from the python ones.

Reply:
Ah since the Python changes are related, I figured it would make sense to do it in a single PR. I just created https://github.com/apache/iceberg/pull/5291 for the Java change.